boot/bootutil/src/bootutil_priv.h

```diff
@@ -56,9 +56,10 @@ struct flash_area;
 
 #if (defined(MCUBOOT_OVERWRITE_ONLY) + \
      defined(MCUBOOT_SWAP_USING_MOVE) + \
+     defined(MCUBOOT_SWAP_USING_SCRATCH) + \
      defined(MCUBOOT_DIRECT_XIP) + \
      defined(MCUBOOT_RAM_LOAD)) > 1
-#error "Please enable only one of MCUBOOT_OVERWRITE_ONLY, MCUBOOT_SWAP_USING_MOVE, MCUBOOT_DIRECT_XIP or MCUBOOT_RAM_LOAD"
+#error "Please enable only one of MCUBOOT_OVERWRITE_ONLY, MCUBOOT_SWAP_USING_MOVE, MCUBOOT_SWAP_USING_SCRATCH, MCUBOOT_DIRECT_XIP or MCUBOOT_RAM_LOAD"
```

Comment:
`MCUBOOT_SWAP_USING_SCRATCH` is enabled by default when no other mode is selected, just a few lines below. If you want to also add it to the `#error` message put this `#if` block below the next one.

Reply:
Then the code that defines `MCUBOOT_SWAP_USING_SCRATCH` should be moved out of this file.